request_test.go

```diff
@@ -504,82 +504,212 @@ func TestRequest_URLConcatenate(t *testing.T) {
 
 	reporter := NewAssertReporter(t)
 
-	config1 := Config{
-		BaseURL:  "",
-		Client:   client,
-		Reporter: reporter,
-	}
-
-	config2 := Config{
-		BaseURL:  "http://example.com",
-		Client:   client,
-		Reporter: reporter,
+	cases := []struct {
+		name      string
+		baseURL   string
+		method    string
+		path      string
+		pathArgs  []interface{}
+		setupFunc func(request *Request)
+		fullURL   string
+	}{
+		{
+			name:    "empty url, empty path",
+			baseURL: "",
+			method:  "GET",
+			path:    "",
+			fullURL: "",
+		},
+		{
+			name:    "empty path",
+			baseURL: "http://example.com",
+			method:  "GET",
+			path:    "",
+			fullURL: "http://example.com",
+		},
+		{
+			name:    "slash empty path",
```

Comment:
nit: I suggest "empty path, url with slash" for consistency with other names.

Also maybe rename "with slash" / "without slash" to "url with slash" / "url without slash" for clarify? Otherwise one can think that "with slash" is about path.